Update us_state_populations to include the year 2020

lm-hien opened the pull request:
- Re-downloaded raw data nhgis001-ts-nominal-state.csv
- Updated the script us-state-populations.R as the raw data now is in a different format as before

lm-hien committed:
Update us_state_populations to include the year 2020

lm-hien commented:
@ropensci/admin Hello could someone please review my pull request and perhaps give me some feedback? I'm still trying to figure out if I'm doing it the correct way. Thank you so much!

maelle commented on DESCRIPTION:
```diff
@@ -16,7 +16,7 @@ Description: These sample data sets are intended for historians
     quantitative analysis, and network analysis.
 Depends:
-    R (>= 3.0.2)
+    R (>= 3.5)
```
Why do you need to increase this? (For context https://blog.r-hub.io/2022/09/12/r-dependency/)

lm-hien replied:
Thank you for the feedbacks!

I think usethis automatically detected the new R version dependency from my new data cleaning script file and made that change for me. To be honest, I didn't think much about the R dependency when writing the script; I just used functions that I'm familiar with. What do you think is the good solution right now? Should I rewrite my code in the way that make it still compatible with R 3.0.2?

maelle replied:
No it's fine, it might something related to data compression. R 3.5.0 was released in 2018 so it's absolutely fine to request it. I was just curious. :smile_cat:

maelle commented on DESCRIPTION:
```diff
@@ -16,7 +16,7 @@ Description: These sample data sets are intended for historians
     military, and prosopographical data suitable for mapping,
```
I'd recommend adding an item to the `NEWS.md` file. https://devguide.ropensci.org/maintenance_releases.html#news

maelle replied:
The item could state what has been updated, with a few details (for instance, more years of data?).

lm-hien replied:
Okay thanks for pointing this out. Should I also update the documentation for that dataset to reflect that change?

maelle commented on data-raw/scripts/us-state-populations.R:
```diff
@@ -1,10 +1,20 @@
                                  stringsAsFactors = FALSE)
 us_state_populations <- us_state_populations %>%
+  pivot_longer(
+    cols = matches("A00AA\\d{4}$"),
```
I think a comment or an explaining variable might help future readers. For instance

```r
pattern_of_blabla <- "A00AA\\d{4}"
us_state_populations <- us_state_populations %>%
  pivot_longer(
    cols = matches(sprintf("%s$", pattern_of_blabla)),
    names_to = "year",
    values_to = "population",
    names_pattern = pattern_of_blabla
  ) %>%
```

where "blabla" would actually explain what the pattern recognizes.

lm-hien replied:
I can do that! Thanks for pointing that out!

maelle commented on data-raw/nhgis0001_ts_nominal_state.csv:
```diff
@@ -0,0 +1,85 @@
+GISJOIN,STATE,STATEFP,STATENH,A00AA1790,A00AA1800,A00AA1810,A00AA1820,A00AA1830,A00AA1840,A00AA1850,A00AA1860,A00AA1870,A00AA1880,A00AA1890,A00AA1900,A00AA1910,A00AA1920,A00AA1930,A00AA1940,A00AA1950,A00AA1960,A00AA1970,A00AA1980,A00AA1990,A00AA2000,A00AA2010,A00AA2020
```
what are those new files?

lm-hien replied:
They are raw data I got from the same source. They have slightly different columns which I had to rewrite the cleaning script for. I couldn't find the raw data with the exact same column format from that source. I don't think they offer that anymore.

maelle commented on data-raw/scripts/us-state-populations.R:
```diff
@@ -1,10 +1,20 @@
 library(dplyr)
-us_state_populations <- read.csv(file.path("data-raw", "nhgis0011_ts_state.csv"),
```
should the new file be saved under the same name as the former one?

lm-hien replied:
Can you clarify a little bit on this? Thank you so much!

maelle replied:
I wondered whether it'd make sense to save the *new* raw data file under `nhgis0011_ts_state.csv` (the old name) but it might make no sense at all. :grin:

maelle reviewed (commented):
Thank you!

lm-hien committed:
- Update NEWS.md as well as the roxygen comments in R/us-state-populations.R to reflect the changes
- Get rid of the old raw data files
- Create a more explainatory name for the Regex pattern in data-raw/scripts/us-state-populations.R

maelle commented on R/us-state-populations.R:
```diff
@@ -1,4 +1,4 @@
-#' Populations of US states and territories, 1790-2010
+#' Populations of US states and territories, 1790-2020
```
so the previous update had not updated the year? Since you only added 2020.

lm-hien replied:
This dataset only contains data for every ten years so it goes 1790, 1800, 1810... 2010, 2020. The previous update was correct!

maelle replied:
aaaah ok!!

maelle commented on R/us-state-populations.R:
```diff
@@ -16,11 +16,13 @@
 #'
 #'   }
-#' @format A data frame with 983 observations of 4 variables.
+#' @format A data frame with 1034 observations of 4 variables.
```
Since `devtools::document()` loads the package would it work to do this:

```suggestion
#' @format A data frame with `r nrow(us_state_populations)` observations of 4 variables.
```

lm-hien replied:
Thanks for the suggestion! I'll work on that.

lm-hien replied:
So the code above doesn't work. And I haven't seen any examples where the number of rows is rendered dynamically: https://github.com/search?q=%22A+data+frame+with%22+%22rows%22+%22variables%22+owner%3Acran+language%3AR&type=code&ref=advsearch. So I've decided to hard code it for now. Thanks for the suggestions tho!

maelle replied:
Thanks for trying!

maelle commented on R/us-state-populations.R:
```diff
@@ -16,11 +16,13 @@
+#'   Steven Manson, Jonathan Schroeder, David Van Riper, Katherine Knowles, Tracy Kugler,
+#'   Finn Roberts, and Steven Ruggles.
+#'   \emph{IPUMS National Historical Geographic Information System: Version 19.0}
+#'   [dataset]. Minneapolis, MN: IPUMS, 2024. http://doi.org/10.18128/D050.V19.0.
```
For DOI there's `\doi{` https://github.com/search?q=%2F%5C%5Cdoi%5C%7B%2F+owner%3Acran+language%3AR&type=code&ref=advsearch&l=R

lm-hien replied:
Oh, thanks. I didn't know this before!

maelle reviewed (commented):
Great work :rocket:

lm-hien committed:
- Fix the DOI link and add the escape for the squared brackets.

maelle commented on man/us_state_populations.Rd:
```diff
@@ -5,7 +5,7 @@
 \title{Populations of US states and territories, 1790-2020}
 \format{
-A data frame with 1034 observations of 4 variables.
+A data frame with `nrow(us_state_populations)` observations of 4 variables.
```
you need to re-run `document()`

maelle reviewed (approved):
Just a small comment!

lm-hien committed:
Rerun devtools::document()
